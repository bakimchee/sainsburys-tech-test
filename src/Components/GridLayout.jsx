import React from 'react';
import styled from 'styled-components';
import { theme } from '../Theme';

const GridWrapper = styled.div`
  display: block;
  padding: 0 ${theme.size.spacing.large};
  max-width: 960px;
  margin: 0 auto;
`;

const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(2, 1fr);
  grid-column-gap: ${theme.size.spacing.default};
  grid-row-gap: ${theme.size.spacing.default};
`;

const GridLayout = ({ children }) => (
  <GridWrapper id="main">
    <Grid>{children}</Grid>
  </GridWrapper>
);

export default GridLayout;
