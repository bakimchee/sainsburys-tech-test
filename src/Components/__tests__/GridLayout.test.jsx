/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import GridLayout, { Grid } from '../GridLayout';

test('Grid', () => {
  const component = renderer.create(<Grid>Component</Grid>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('display', 'grid', {
    media: '(min-width: 640px)',
  });
});

test('Grid Layout', () => {
  const component = renderer.create(<GridLayout>Grid</GridLayout>);
  const tree = component.toJSON();
  expect(tree.props.id).toEqual('main');
});
