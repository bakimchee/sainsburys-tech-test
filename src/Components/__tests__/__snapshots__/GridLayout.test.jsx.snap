// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid 1`] = `
@media (min-width:640px) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(4,1fr);
    grid-template-rows: repeat(2,1fr);
    grid-column-gap: 8px;
    grid-row-gap: 8px;
  }
}

<div
  className="c0"
>
  Component
</div>
`;
